### PR TITLE
[Amplify UI Components] - Docs for chatbot components

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:*

*Description of changes: This adds documentation for the new chatbot ui, which is currently on preview.

Note that due to existing build errors with `aws-sdk@gamma-v4` on latest amplify-js, prop list is typed manually. This will be replaced in favor of `<ui-component-props />` after `aws-sdk` version has been bumped. 

Here's what the list looks like on React:

Screen Shot 2020-08-25 at 11 08 13 PM

and on Vue, Angular, and Ionic:

Screen Shot 2020-08-25 at 11 08 25 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/ui/interactions/fragments/web/migration.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
++ providers: [],
+  bootstrap: [AppComponent],
+})
+export class AppModule {}
+```
+
+_app.component.html_
+
+```diff
++ <amplify-chatbot bot-name="yourBotName/>
```
Comment (sammartinez) [inline docs/ui/interactions/fragments/web/migration.md:54]:
```suggestion
+ <amplify-chatbot bot-name="yourBotName" />
```

File: docs/ui/interactions/fragments/web/migration.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
```
Comment (sammartinez) [inline docs/ui/interactions/fragments/web/migration.md:88]:
```suggestion
+ <amplify-chatbot bot-name="yourBotName" />
```

File: docs/ui/interactions/fragments/web/usage.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+</template>
+```
+</docs-filter>
+
+<ui-component-props tag="amplify-chatbot" use-table-headers></ui-component-props>
+
+## Use Cases
+
+### Setting Up Voice Chat
+In order for voice input to work with Amazon Lex, you may have to enable output voice in [AWS Console](https://console.aws.amazon.com/console/home). Under the Amazon Lex service, click on your configured Lex chatbot and go to settings -> General and pick your desired output voice.
```
Comment (sammartinez) [inline docs/ui/interactions/fragments/web/usage.md:127]:
```suggestion
In order for voice input to work with Amazon Lex, you may have to enable output voice in [AWS Management Console](https://console.aws.amazon.com/console/home). Under the Amazon Lex service, click on your configured Lex chatbot and go to settings -> General and pick your desired output voice. 
```

File: docs/ui/interactions/fragments/web/installation.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+## Installation
+
+<docs-filter framework="react">
+
+```
+yarn add aws-amplify @aws-amplify/ui-react
```
Comment (sammartinez) [inline docs/ui/interactions/fragments/web/installation.md:6]: If we are looking to use this on the `preview-tag` we will probably want to have this point to the specific version to place this into.

File: docs/ui/interactions/fragments/web/chatbot.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+A simple way to add a conversational UI into your app is to use our ChatBot Component.
+
+ChatBot automatically renders a complete chat messaging interface that can be used out-of-the-box, or it can be customized using theming support.
+
+<inline-fragment src=~/ui/interactions/fragments/web/installation.md></inline-fragment>
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/chatbot.md:5]: Let's add a callout section here describing the component's preview state.

File: docs/ui/interactions/fragments/web/installation.md (hunk 4 above)
Comment (jordanranz) [reply docs/ui/interactions/fragments/web/installation.md:6]: Yep, I agree^

File: docs/ui/interactions/fragments/web/customization.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+## Customization
+
+### Slots
+
+Amplify UI Components provides the following [slot](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot).
+
+| Name                 | Description                         |
+| -------------------- | ------------------------------------|
+| `Header`             | Content placed at the top of the UI |
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/customization.md:9]: For now this works. We have access to this documentation via the docs.json: https://github.com/aws-amplify/docs/blob/0682ec157a414cc6bfe096979be4d652f9fd2c9a/client/src/docs-ui/ui-component-props/ui-component-props.tsx#L2 but we can save this for our GA launch

File: docs/ui/interactions/fragments/web/customization.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+### Custom CSS Properties
+
+`amplify-chatbot` provides the following CSS properties to modify the style at component level.
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/customization.md:13]: Same comment as above

File: docs/ui/interactions/fragments/web/chatbot.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+<inline-fragment src="~/ui/interactions/fragments/web/usage.md"></inline-fragment>
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/chatbot.md:7]: These fragments seem pretty chatbot specific, can we put them in a sub folder? `web/chatbot/usage.md`

File: docs/ui/interactions/fragments/web/installation.md (hunk 4 above)
Comment (wlee221) [reply docs/ui/interactions/fragments/web/installation.md:6]: On it

File: docs/ui/interactions/fragments/web/chatbot.md (hunk 5 above)
Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot.md:5]: Done

File: docs/ui/interactions/fragments/web/chatbot.md (hunk 8 above)
Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot.md:7]: Sure

File: docs/ui/interactions/fragments/web/chatbot/usage.md [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+    bot-name="yourBotName"
+    bot-title="My ChatBot"
+    welcome-message="Hello, how can I help you?"
+  />
+</template>
+```
+
+</docs-filter>
+
+<inline-fragment framework="react" src="~/ui/interactions/fragments/web/chatbot/properties-jsx.md"></inline-fragment>
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/chatbot/usage.md:128]: I remember there was a limitation to a nested `inline-fragment` like this. Is this working properly?

Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot/usage.md:128]: Yes, this was working properly on local.

File: docs/ui/interactions/fragments/web/chatbot.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+ChatBot component is currently published on `ui-preview` tag for customer feedback.
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/chatbot.md:1]:
```suggestion
<amplify-callout warning>
The Chatbot component is currently in Preview and published on `@ui-preview` tag for customer feedback.
</amplify-callout>
```

File: docs/ui/interactions/fragments/web/chatbot.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+[![version](https://img.shields.io/npm/v/@aws-amplify/ui-react/ui-preview.svg)](https://www.npmjs.com/package/@aws-amplify/ui-react)
```
Comment (jordanranz) [inline docs/ui/interactions/fragments/web/chatbot.md:5]: This is React specific. We can maybe wrap this is a `docs-filter` for other frameworks.

Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot.md:5]: I'll wrap it

File: docs/ui/interactions/fragments/web/chatbot.md (hunk 10 above)
Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot.md]: 👍

File: docs/ui/interactions/fragments/web/chatbot.md (hunk 11 above)
Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot.md:7]: Done!

File: docs/ui/interactions/fragments/web/chatbot/migration.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+To migrate from the legacy chatbot component to the latest chatbot component, use the steps below:
+
+### Usage
+
+<docs-filter framework="react">
+
+```diff
+- import { ChatBot } from 'aws-amplify-react';
++ import { AmplifyChatbot } from '@aws-amplify/ui-react';
```
Comment (sammartinez) [inline docs/ui/interactions/fragments/web/chatbot/migration.md:11]: Can you look to make sure the diffs are pointing to the preview as well for all below

Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot/migration.md:11]: nice catch, thank you

Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot/migration.md:11]: I'll append a comment to each of `/ui-<framework>` then.

Comment (wlee221) [reply docs/ui/interactions/fragments/web/chatbot/migration.md:12]: done!

**Comment by wlee221** [comment]
Also pushed a change to not use self closing element on angular/ionic

**Comment by wlee221** [comment]
ty @jordanranz @sammartinez!

**Pull request merged**